### ISSUE TITLE
This utility needs better documentation

### ISSUE THREAD

**Issue by dnkmmr69420** [numtide/system-manager#21]
Can this tool have better documentation please. I have no idea how to use this tool

**Comment by a-h** [comment]
I've built a walkthrough of using system-manager to deploy a simple service on local virtual machines at https://github.com/a-h/system-manager-test/

It uses Canonical's `multipass` to run Ubuntu VMs. The README contains a series of steps used to deploy the example service to a remote virtual machine.

@numtide - I could turn this into a walkthrough and port the code over to this repo where it can be maintained along with the product?

I tested it on an x86-64 Linux machine, and also from my M1 Mac, where Multipass creates aarch64 Linux VMs. The only change required was to set this field https://github.com/a-h/system-manager-test/blob/33765ac34b9f3b25f16098a565d0cadd8903d5eb/machines/worker-001/modules/default.nix#L5 - I expect it's possible for that to be a parameter, but wasn't sure on the best way.

**Comment by ShalokShalom** [comment]
I think the README is missing the path to the config file. 
Am I supposed to use `/etc/nix/nix.conf`?

I will add that, then.

**Comment by OJarrisonn** [comment]
Agreed, i'm looking for using this so i can install docker using nixpkgs on my debian machine but didn't understood so far how to use it

**Comment by zimbatm** [comment]
For some reason, this issue has escaped my attention.

We have other priorities at the moment, but I added this to our backlog.

@a-h absolutely. I'm not sure exactly what the issue is with `nixpkgs.hostPlatform` but we can work this out if you send a PR.

**Comment by Endle** [comment]
Can we add a list of examples, so current system-manager users can share their nix config files?

**Comment by eureka-cpu** [comment]
> Agreed, i'm looking for using this so i can install docker using nixpkgs on my debian machine but didn't understood so far how to use it

Hey @OJarrisonn I'm just going through the currently open issues and saw your comment. As far as I know, that's not something that's currently possible, as stated here in the [NixOS Docker Wiki](https://wiki.nixos.org/wiki/Docker#Running_the_docker_daemon_from_nix-the-package-manager_-_not_NixOS).

**Comment by phanirithvij** [comment]
https://discourse.nixos.org/t/how-to-run-docker-daemon-from-nix-not-nixos/43413/5

https://nixos.wiki/wiki/Docker#Running_the_docker_daemon_from_nix-the-package-manager_-_not_NixOS

**Comment by eureka-cpu** [comment]
Ah fantastic, I'll verify and add this to the docs.

**Comment by jfroche** [comment]
Do not hesitate to reopen if the last documentation updates doesn't help you.
